cmd/staterecovery/staterecovery.go

```diff
@@ -0,0 +1,85 @@
+package staterecovery
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/trie"
+	"github.com/ethereum/go-ethereum/trie/triedb/hashdb"
+)
+
+func RecreateMissingStates(chainDb ethdb.Database, bc *core.BlockChain, cacheConfig *core.CacheConfig) error {
```

Comment:
Instead of the option just being a RecreateMissingState bool, let's make it a RecreateMissingStateFrom parameter that takes a block number to start from. That way if the process gets interrupted a user can recover it manually.

Reply:
Also, we shouldn't try to recreate state for classic blocks. If the user passes in a block number lower than the chain config's ArbitrumChainParams GenesisBlockNumber, we should emit a warning and start from that genesis block instead. That's not the same as what the blockchain says is the genesis, which is always block 0.

Reply:
good idea, thanks! I applied the change